src/util/orm.py

```diff
@@ -424,6 +426,37 @@ def custom_module_field_as_manual(env, rollback=True, do_flush=False):
         )
         mock_x_email_fields = [r[0] for r in env.cr.fetchall()]
 
+    temp_ir_model_access_ids = []
+    if version_gte("14.0") and custom_models:
+        # 1.2 Version 14.0 and above requires new access rights for transient models,
+        # and the rights are yet to be added post standard upgrade for custom modules.
+        # To avoid access errors, create some temporary access rights for these models.
+        env.cr.execute(
+            """
+                SELECT id
+                  FROM ir_model
+                 WHERE model IN %s
+                   AND transient
+                   AND id NOT IN (
+                           SELECT DISTINCT model_id
+                             FROM ir_model_access
+                            WHERE active
+                       )
+            """,
+            [custom_models],
+        )
+
+        no_access_custom_transient_ids = env.cr.fetchall()
+
+        if no_access_custom_transient_ids:
+            insert_query = """
```

Comment:
this should be unified into the previous query (with a cte), return only the created ids.

Reply:
indeed looks much better, thanks